src/combine_raw_datasets.py

```diff
@@ -1,9 +1,23 @@
-from core_data_modules.traced_data import TracedData
+from core_data_modules.traced_data import TracedData, Metadata
+from core_data_modules.util import TimeUtils
 
 
 class CombineRawDatasets(object):
     @staticmethod
-    def combine_raw_datasets(user, messages_datasets, follow_up_survey_datasets, demogs_datasets):
+    def coalesce_traced_runs_by_key(user, traced_runs, coalesce_key):
+        coalesced_runs = dict()
+
+        for run in traced_runs:
+            if run[coalesce_key] not in coalesced_runs:
+                coalesced_runs[run[coalesce_key]] = run
+            else:
+                coalesced_runs[run[coalesce_key]].append_data(
+                    dict(run.items()), Metadata(user, Metadata.get_call_location(), TimeUtils.utc_now_as_iso_string()))
+
+        return list(coalesced_runs.values())
+
+    @staticmethod
+    def combine_raw_datasets(user, messages_datasets, follow_up_survey_datasets, coalesced_demog_datasets):
```

Comment:
Can you call these: messages_datasets, demog_datasets, and follow_up_[survey_]datasets?
Or: messages_datasets, coalesced_demog_datasets, coaleseced_follow_up_[survey_]datasets

(Both the follow_ups and the demogs should have gone through the coalesce routine)